orderBook.js

```diff
@@ -1,5 +1,52 @@
-function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+function getUpdatedBook(existingBook, incomingOrder) {
+  let updatedBook = []
+  let correspondingOrders = existingBook.filter((item) => {
+    return item.type === incomingOrder.type
+  })
+
+  if (correspondingOrders.length === existingBook.length) {
```

Comment:
Not sure what this check is intended to do. They'd only have an equal length if all orders matched or no orders matched.